applications/protocols_serialization/server/src/main.c

```diff
@@ -16,6 +17,13 @@ static void err_handler(const struct nrf_rpc_err_report *report)
 	//k_oops();
 }
 
+static void button_handler(uint32_t button_state, uint32_t has_changed)
+{
+	if (button_state & DK_ALL_BTNS_MSK) {
+		k_oops();
```

Comment:
Seems this is testing stuff that shouldn't be merged.

Reply:
How will this feature be tested if we don't provide such an trigger?

Reply:
I guess there is misunderstanding of the ncs-sdk application purposes.
Application is logically completed functionality that is ready for being used by Nordic's customers `as is` for their production\deployment and etc `from the box`.

This is not for testing, not even for the feature demonstration (for demonstration purposes there are samples).

It is why I'm confused by seeing a lot of garbage code, inaccurate configuration with huge buffers (without clear explanation of the reason of them) etc.

All functionality that is added within the serialization project should (shall?) be tested separately in form of system\unit\performance\robustness tests implemented in the test environment and integrated in CI.

Application should be cleaned up from any testing stubs and stuffs.
If the initially it was assumed for testing then application is wrong place for it. Probably we do not need application at all and just a couple of samples will be sufficient, for example for demonstration crash logging.

Reply:
I'm not sure what you mean by the garbage code and huge buffers etc. I thought it's clear that while the server can satisfy your requirements of an application to some extent, the client is just an exemplary code because we need some way to validate the server's functionality.

I personally don't feel that turning the client into a sample in unrelated directory would make it more logical for our customers but feel free to bring this up at the stand up meeting. Eventually, this is up to PMT what exactly they want to be delivered.

I get your point that an application (let's confine this to our server) should be optimized etc. but I don't think we have to artificially copy its functionality into a bunch of samples just to satisfy the requirements that "applications" shouldn't be used for testing. I don't think this is true even today - Thingy:53 applications do have dedicated integration or nightly tests.

And if we add extra snippets to enable debug features, what's exactly the harm for the customers in deploying the application in production (without enabling the debug features)?

Reply:
Sure, I'll raise up the question on stand up. I'm really not a big fan of maintaining some code that looks like `all inclusive` just for testing purposes.

That the application exists for not testing purposes doesn't mean the application shouldn't be tested by self. I do not see any contradiction in testing application functionality as you pointed for some platforms. Even more, listed above type of tests should include application testing to not do it manually before every release.

Also, there is no contradiction to have debug logging(in form of snippet for example) in the final product if a customer deployed the network in test mode to gather statistics or figure out some instabilities.